pkg/crc/image/image.go

```diff
@@ -0,0 +1,129 @@
+package image
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	crcpreset "github.com/code-ready/crc/pkg/crc/preset"
+	"github.com/code-ready/crc/pkg/crc/version"
+	"github.com/code-ready/crc/pkg/extract"
+	"github.com/containers/image/v5/copy"
+	"github.com/containers/image/v5/directory"
+	"github.com/containers/image/v5/docker"
+	"github.com/containers/image/v5/signature"
+	"github.com/containers/image/v5/types"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const (
+	registryURI = "quay.io"
+	orgName     = "crcont"
+)
+
+type imageHandler struct {
+	imageName string
+	imageTag  string
+}
+
+func (img *imageHandler) uri() string {
+	uri := url.URL{
+		Host: registryURI,
+		Path: orgName + "/" + img.imageName + ":" + img.imageTag,
+	}
+	return uri.String()
+}
+
+func (img *imageHandler) policyContext() (*signature.PolicyContext, error) {
+	policy := &signature.Policy{Default: []signature.PolicyRequirement{signature.NewPRInsecureAcceptAnything()}}
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return nil, fmt.Errorf("error creating security context: %s", err.Error())
+	}
+
+	return policyContext, nil
+}
+
+func (img *imageHandler) getSystemContext() *types.SystemContext {
+	return &types.SystemContext{
+		OSChoice:           runtime.GOOS,
+		ArchitectureChoice: runtime.GOARCH,
+	}
+}
+
+// copyImage pull the image from the registry and put it to destination path
+func (img *imageHandler) copyImage(destPath string, reportWriter io.Writer) (*v1.Manifest, error) {
+	// Source Image from docker transport
+	srcImg := img.uri()
+	srcRef, err := docker.ParseReference(srcImg)
+	if err != nil {
+		return nil, fmt.Errorf("invalid source image name %s: %v", srcImg, err)
+	}
+
+	destRef, err := directory.Transport.ParseReference(destPath)
+	if err != nil {
+		return nil, fmt.Errorf("invalid destination name %s: %v", destPath, err)
+	}
+
+	policyContext, err := img.policyContext()
+	if err != nil {
+		return nil, err
+	}
+
+	manifestData, err := copy.Image(context.Background(), policyContext,
+		destRef, srcRef, &copy.Options{
+			ReportWriter: reportWriter,
+			SourceCtx:    img.getSystemContext(),
+		})
+	if err != nil {
+		return nil, err
+	}
+	manifest := &v1.Manifest{}
+	if err := json.Unmarshal(manifestData, manifest); err != nil {
+		return nil, err
+	}
+	return manifest, nil
+}
+
+func getLayerPath(m *v1.Manifest) string {
+	return strings.TrimPrefix(m.Layers[0].Digest.String(), "sha256:")
+}
+
+func getImageName(preset crcpreset.Preset) string {
+	if preset == crcpreset.Podman {
+		return "podman-bundle"
```

Comment:
Can these be extracted as part of a constants file, like named `podmanBundleName` and `openshiftBundleName`. Pretty sure these will be somehow reused of perhaps even overwritten at some point.

Reply:
These are images name not the bundle so not sure if that is reused anywhere except in image package.